src/PrivateDns/PrivateDns/Zones/SetAzurePrivateDnsZone.cs

```diff
@@ -0,0 +1,104 @@
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The full name of the zone (without a terminating dot).", ParameterSetName = FieldsParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group in which the zone exists.", ParameterSetName = FieldsParameterSetName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = FieldsParameterSetName)]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = ResourceParameterSetName)]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The zone object to set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PrivateDnsZone Zone { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ObjectParameterSetName)]
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ResourceParameterSetName)]
+        public SwitchParameter Overwrite { get; set; }
+
+        [Parameter( ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipeline = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+
+            PrivateDnsZone result = null;
+            PrivateDnsZone zoneToUpdate = null;
+
+            if (this.ParameterSetName == FieldsParameterSetName || this.ParameterSetName == ResourceParameterSetName)
+            {
+                if (!string.IsNullOrEmpty(this.ResourceId))
+                {
+                    PrivateDnsUtils.GetResourceGroupNameAndZoneNameFromResourceId(this.ResourceId, out var resourceGroupName, out var zoneName);
+                    this.ResourceGroupName = resourceGroupName;
+                    this.Name = zoneName;
+                }
+
+                if (this.Name.EndsWith("."))
+                {
+                    this.Name = this.Name.TrimEnd('.');
+                    this.WriteWarning($"Modifying zone name to remove terminating '.'.  Zone name used is \"{this.Name}\".");
+                }
+
+                zoneToUpdate = this.PrivateDnsClient.GetPrivateDnsZone(this.Name, this.ResourceGroupName);
+                zoneToUpdate.Etag = "*";
```

Comment:
Why are you setting ETag to "*" when you are taking an explicit Overwrite param in the cmdlet for the Resource param set?

Reply:
this was a mistake, overwrite param is only for object param sets  and not for resourceidparam sets.